File: app/src/main/java/de/wollis_page/gibsonos/model/Account.java
package de.wollis_page.gibsonos.model;

import android.os.Parcel;
import android.os.Parcelable;

import com.orm.SugarRecord;

import java.util.Objects;

import androidx.annotation.NonNull;


public class Account extends SugarRecord implements Parcelable
{
    public static final String EXTRA_ACCOUNT = "account";

    private String alias;
    private String user;
    private String token;
    private String url;

    public Account()
    {
    }

    public Account(@NonNull String user, @NonNull String url, String token)
    {
        setUser(user);
        setToken(url);
        setToken(token);
    }

    public Account(@NonNull String user, @NonNull String url, String token, String alias)
    {
        setUser(user);
        setUrl(url);
        setToken(token);
        setAlias(alias);
    }

    private Account(Parcel in) {
        readFromParcel(in);
    }

    public String getAlias() {
        return alias;
    }

    public void setAlias(String alias) {
        this.alias = alias;
    }

    public String getUser() {
        return user;
    }

    public void setUser(@NonNull String user) {
        this.user = user;
    }

    public String getToken() {
        return token;
    }

    public void setToken(String token) {
        this.token = token;
    }

    public String getUrl() {
        return url;
    }

    public void setUrl(@NonNull String url) {
        this.url = url;
    }

    public void writeToParcel(Parcel dest, int flags) {
        dest.writeLong(getId());
        dest.writeString(getAlias());
        dest.writeString(getUser());
        dest.writeString(getToken());
        dest.writeString(getUrl());
    }

    private void readFromParcel(Parcel in) {
        setId(in.readLong());
        setAlias(in.readString());
        setUser(Objects.requireNonNull(in.readString()));
        setToken(in.readString());
        setUrl(Objects.requireNonNull(in.readString()));
    }

    public static final Parcelable.Creator<Account> CREATOR = new Parcelable.Creator<Account>() {
        public Account createFromParcel(Parcel in) {
            return new Account(in);
        }

        public Account[] newArray(int size) {
            return new Account[size];
        }
    };

    public int describeContents() {
        return 0;
    }
}

File: app/src/main/java/de/wollis_page/gibsonos/model/Account.kt
package de.wollis_page.gibsonos.model

import android.os.Parcel
import android.os.Parcelable
import android.util.Log
import com.orm.SugarRecord
import de.wollis_page.gibsonos.dto.ListItemInterface
import de.wollis_page.gibsonos.helper.Config

class Account: SugarRecord, Parcelable, ListItemInterface {
    var alias: String? = null
    var userId: Long = 0
    lateinit var userName: String
    var token: String? = null
    lateinit var url: String
    var deviceId: Long = 0

    constructor()

    constructor(userId: Long, userName: String, deviceId: Long, url: String, token: String?) {
        Log.i(Config.LOG_TAG, "Create Account Model $userName@$url with token $token")

        this.userId = userId
        this.userName = userName
        this.url = url
        this.token = token
        this.deviceId = deviceId
    }

    private constructor(parcel: Parcel) {
        this.id = parcel.readLong()
        this.alias = parcel.readString()
        this.userId = parcel.readLong()
        this.userName = parcel.readString().toString()
        this.token = parcel.readString()
        this.url = parcel.readString().toString()
        this.deviceId = parcel.readLong()
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeLong(this.id)
        dest.writeString(this.alias)
        dest.writeLong(this.userId)
        dest.writeString(this.userName)
        dest.writeString(this.token)
        dest.writeString(this.url)
        dest.writeLong(this.deviceId)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<Account> {
        override fun createFromParcel(parcel: Parcel): Account {
            return Account(parcel)
        }

        override fun newArray(size: Int): Array<Account?> {
            return arrayOfNulls(size)
        }
    }
}